resources/2024-07-secretstore-feature-spec.md

```diff
@@ -0,0 +1,426 @@
+  name: 'demo'
+  properties: {
+    application: application
+    container: {
+      image: 'ghcr.io/radius-project/samples/demo:latest'
+      env:{
++        DB_CONNECTION: {
++          valueFrom: {
++            secretRef: {
++              source: authcreds.id
++              key: 'username'
++            }
++          }
++        }
+      }
+      ports: {
+        web: {
+          containerPort: 3000
+        }
+      }
+    }
+  }
+}
+```
+
+(b) Developer references the `Applications.Core/secretStores` resource in their `Applications.Core/containers` resource definition under the `volumes` object within the `app.bicep` application definition to write the secret into a file on disk/memory with a specified file path and name, then mount that volume into their container at deploy time.
+
+```diff
+resource demo 'Applications.Core/containers@2023-10-01-preview' = {
+  name: 'demo'
+  properties: {
+    application: application
+    container: {
+      image: 'ghcr.io/radius-project/samples/demo:latest'
+      ports: {
+        web: {
+          containerPort: 3000
+        }
+      }
+      volumes: {
+        ephemeralVolume: {
+          kind: 'ephemeral'
+          mountPath: '/tmpfs'
+          managedStore: 'memory'
++         secrets: {
++           filePath: '/path/to/secret' // default is /var/run/secrets/
++           fileName: 'secretfile.txt' // default is secretfile.txt
++           username: {
++             valueFrom: {
++               secretRef: {
++                 source: authcreds.id
++                 key: 'username'
++               }
++             }
++           }
++         }
+        }
+        persistentVolume: {
+          kind: 'persistent'
+          source: volume.id
++         secrets: {
++           filePath: '/path/to/secret' // default is /var/run/secrets/
++           fileName: 'secretfile.txt' // default is secretfile.txt
++           username: {
++             valueFrom: {
++               secretRef: {
++                 source: authcreds.id
++                 key: 'username'
++               }
++             }
++           }
++         }
+        }
+    }
+  }
+}
+```
+> Note: the`secrets`, `filePath`, `fileName` properties and default values are just proposals and may be changed during tech design. They should also be optional for the `volumes` property in `Applications.Core/containers` to write the secret into a file on disk/memory with a specified file path and name, with the proposed defaults being `/var/run/secrets/` and `secretfile.txt` respectively.
+
+(c) Developer references the `Applications.Core/secretStores` resource in their `Applications.Extenders`, `Applications.Datastores/*`, `Applications.Messaging/*`, or `Applications.Dapr/*` resource definition to securely manage secrets for use in their application. Radius then uses the secret to authenticate into the resource at deploy time. The secrets might be referenced in the resources within the `app.bicep` application definition as follows:
```

Comment:
> Radius then uses the secret to authenticate into the resource at deploy time.

Apologies if this was already discussed during discussion - I didn't fully understand what do we mean by Radius authenticating into the resource?